x-pack/plugins/security_solution/public/common/components/top_n/helpers.ts

```diff
@@ -65,3 +68,82 @@ export const getOptions = (activeTimelineEventsType?: TimelineEventsType): TopNO
       return defaultOptions;
   }
 };
+
+/** returns true if the specified timelineId is a detections alert table */
+export const isDetectionsAlertsTable = (timelineId: string | undefined): boolean =>
+  timelineId === TimelineId.detectionsPage || timelineId === TimelineId.detectionsRulesDetailsPage;
+
+/**
+ * The following fields are used to filter alerts tables, (i.e. tables in the
+ * `Security > Alert` and `Security > Rule > Details` pages). These fields,
+ * MUST be ignored when showing Top N alerts for `raw` documents, because
+ * the raw documents don't include them.
+ */
+export const IGNORED_ALERT_FILTERS = [
```

Comment:
nit: might make sense to use the consts from '@kbn/rule-data-utils' here and in the tests above just in case these are changed en masse again

Reply:
great suggestion @kqualters-elastic! implemented in https://github.com/elastic/kibana/pull/121562/commits/eea414d8382fe31dd64c1f4b8aaa77a52aec04cf